src/pages/Index.js

```diff
@@ -65,54 +129,189 @@ modalWithImage.setEventListeners();
+      editAvatarModal.renderLoading(false);
+    });
+}
+
+const editAvatarModal = new ModalWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
 );
+editAvatarModal.setEventListeners();
 
-cardSection.renderItems();
+profileAvatarContainer.addEventListener("click", () => {
+  editAvatarModal.open();
+});
 
 /*                                       */
-/*             Event Handlers            */
+/*          LIKE & DISLIKE               */
 /*                                       */
 
+function handleLikeClick(card) {
+  if (card.isLiked === true) {
+    api
+      .removeLike(card._id)
+      .then(() => {
+        card.isLiked = false;
+        card.handleLikeIcon();
+      })
+      .catch(console.error);
+  } else {
+    api
+      .addLike(card._id)
+      .then(() => {
+        card.isLiked = true;
+        card.handleLikeIcon();
+      })
+      .catch(console.error);
+  }
+}
+// function handleLikeClick(card) {
+//   console.log(card);
+//   if (card.isLiked) {
+//     api
+//       .removeLike(card._id)
+//       .then(() => {
+//         card.handleLikeIcon();
+//         card.isLiked = false;
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   }
+//   if (!card.isLiked) {
+//     api
+//       .addLike(card._id)
+//       .then(() => {
+//         card.handleLikeIcon();
+//         card.isLiked = true;
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   }
+// }
+
 function handleImageClick(name, link) {
   modalWithImage.open(name, link);
 }
 
-const user = new UserInfo("#modal-title-input", "#modal-description-input");
+/*                                       */
+/*              ADD CARD                 */
+/*                                       */
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.title, description: data.description });
-  editModal.close();
+function handleAddCardSubmit(data) {
+  const name = data.title;
+  const link = data.link;
+  cardModal.renderLoading(true);
+  api
+    .createNewCard({ name, link })
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      cardModal.close();
+      addCardForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModal.renderLoading(false);
+    });
 }
 
-function handleAddCardSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.link;
-  const newCard = renderCard({ name, link });
-  cardSection.addItem(newCard);
-  cardModal.close();
+/*                                       */
+/*            DELETE CARD                */
+/*                                       */
+
+const confirmDeleteModal = new ModalWithConfirm("#delete-modal");
+confirmDeleteModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  confirmDeleteModal.open();
+  confirmDeleteModal.handleDelete(() => {
+    console.log(card);
```

Comment:
Needs correction: don't forget to notify the user about the loading process by changing the submit button text

Reply:
Still needs correction